tests/test_pay.py

```diff
@@ -1271,8 +1271,11 @@ def test_forward_stats(node_factory, bitcoind):
     l1, l2, l3, l4, l5 = node_factory.get_nodes(5, opts=[{}] * 4 + [{'may_fail': True}])
     node_factory.join_nodes([l1, l2, l3], wait_for_announce=False)
     l2.openchannel(l4, 10**6, wait_for_announce=False)
-    l2.openchannel(l5, 10**6, wait_for_announce=True)
+    l2.openchannel(l5, 10**6, wait_for_announce=False)
 
+    sync_blockheight(bitcoind, [l1, l2, l3, l4, l5])
+    bitcoind.generate_block(1)
+    sync_blockheight(bitcoind, [l1, l2, l3, l4, l5])
```

Comment:
Why sync before *and* after? I think first one can be omitted.

Reply:
Had a bit of a switching around of line here. The problem was slow chain sync on some nodes (valgrind slowing them down), so we'd blast past the 6 blocks (confirmation + 5 to get to the announce height) on the announcing node, but the node receiving the announcement was still stuck at height - 6, and then they'd complain about a future channel. Syncing after the confirmation should be enough though, so I removed the first one.